src/xb-database-manager.c

```diff
@@ -352,6 +352,79 @@ xb_database_manager_register_stopwords (XbDatabaseManager *self,
   return TRUE;
 }
 
+static XapianDatabase *
+create_database_from_manifest (const char  *manifest_path,
+                               GError     **error_out)
+{
+  JsonParser *parser;
```

Comment:
You are kinda mixing the pre c99 declare at top of function and declare anywhere styles.

I'd prefer just declaring when you first use all these things, but up to you